src/history/historyTracker.ts

```diff
@@ -93,6 +93,50 @@ class HistoryStep {
     this.cursorStart = init.cursorStart || undefined;
     this.marks     = init.marks || [];
   }
+
+  /**
+   * merge collapses individual character changes into larger blocks of changes
+   */
+  public merge(): void {
+    if (this.changes.length < 2) {
+      return;
+    }
+    // create new changes list
+    var merged: DocumentChange[] = [];
+    // iterate through the changes, saving the first outside the loop
+    var iter = (this.changes[Symbol.iterator]());
+    var prev = iter.next().value;
+    for (const change of iter) {
+      if (prev.text.length === 0) {
```

Comment:
this `prev` represents changes that we are going to merge or merging, maybe we can give it a more intuitive name?